WordPressKit/WordPressKit/RemoteActivity.swift

```diff
@@ -0,0 +1,116 @@
+import Foundation
+
+open class RemoteActivity {
+    public let activityID: String
+    public let summary: String
+    public let name: String
+    public let type: String
+    public let gridicon: String
+    public let status: String
+    public let rewindable: Bool
+    public let published: Date?
+    public let actor: RemoteActivityActor?
+    public let object: RemoteActivityObject?
+    public let target: RemoteActivityObject?
+    public let items: [RemoteActivityObject]?
+
+    init(dictionary: [String: AnyObject]) {
+        activityID = dictionary["activity_id"] as? String ?? ""
```

Comment:
I'm seeing a bunch of `?? ""`. If those are optional, they should be `String?`. If a `RemoteActivity` isn't valid without an `activityID`, either make this a conditional initializer or a throwing one (I prefer throwing because you can then have breakpoints on error and figure out which item didn't parse)

Reply:
Updated to a throwing initializer to require the `activityId`. For the rest of the fields, I think it makes sense to initialized them as empty.